lib/foreman_inventory_upload/generators/fact_helpers.rb

```diff
@@ -79,7 +79,7 @@ def obfuscate_ips?(host)
         insights_client_setting = ActiveModel::Type::Boolean.new.cast(insights_client_setting)
         return insights_client_setting unless insights_client_setting.nil?
 
-        Setting[:obfuscate_inventory_ips]
+        CommonParameter.find_by(name: 'obfuscate_inventory_ips')&.value
```

Comment:
If it's a parameter, you have to look up the value on the host object itself: https://github.com/theforeman/foreman_rh_cloud/blob/48f9d74ba8a671c74cec3a79b2646c38172a42a3/app/controllers/insights_cloud/api/machine_telemetries_controller.rb#L87